Description: Correct query for formats supported by wxDataObject in wxGTK wxClipboard.
Author: Timothy Lee
Bug: http://trac.wxwidgets.org/ticket/11811
Applied-Upstream: http://trac.wxwidgets.org/changeset/67382
Last-Update: 2012-05-31

Index: wxWidgets/branches/WX_2_8_BRANCH/src/gtk/clipbrd.cpp
===================================================================
--- a/src/gtk/clipbrd.cpp (revision 62148)
+++ b/src/gtk/clipbrd.cpp (revision 67382)
@@ -561,8 +561,9 @@
 
     /* get formats from wxDataObjects */
-    wxDataFormat *array = new wxDataFormat[ data.GetFormatCount() ];
-    data.GetAllFormats( array );
-
-    for (size_t i = 0; i < data.GetFormatCount(); i++)
+    const size_t count = data.GetFormatCount(wxDataObject::Set);
+    wxDataFormat* array = new wxDataFormat[count];
+    data.GetAllFormats(array, wxDataObject::Set);
+
+    for (size_t i = 0; i < count; i++)
     {
         wxDataFormat format( array[i] );
